Source/WebCore/platform/graphics/texmap/TextureMapperDamageVisualizer.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE COMPUTER, INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE COMPUTER, INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "TextureMapperDamageVisualizer.h"
+
+#if ENABLE(DAMAGE_TRACKING)
+#include "Damage.h"
+#include "TextureMapper.h"
+#include <wtf/text/StringToIntegerConversion.h>
+#include <wtf/TZoneMallocInlines.h>
+
+namespace WebCore {
+
+WTF_MAKE_TZONE_ALLOCATED_IMPL(TextureMapperDamageVisualizer);
+
+std::unique_ptr<TextureMapperDamageVisualizer> TextureMapperDamageVisualizer::create()
+{
+    if (const auto* showDamageEnvvar = getenv("WEBKIT_SHOW_DAMAGE")) {
+        const auto value = parseInteger<unsigned>(StringView::fromLatin1(showDamageEnvvar));
+        if (value && *value > 0)
+            return std::unique_ptr<TextureMapperDamageVisualizer>(new TextureMapperDamageVisualizer(*value - 1));
```

Comment:
```suggestion
            return std::unique_ptr<TextureMapperDamageVisualizer>(new TextureMapperDamageVisualizer(*value - 1));
```
```suggestion
            return std::unique_ptr<TextureMapperDamageVisualizer>(new TextureMapperDamageVisualizer(*value - 1));
```
```suggestion
            return WTF::makeUnique<TextureMapperDamageVisualizer>(*value - 1);
```

Reply:
Not possible as `WTF::makeUnique` is not a `friend` of `TextureMapperDamageVisualizer`. I've checked few places and it looks like avoiding befriending is preferred in WebKit

Reply:
Just make the constructor public

Reply:
> Just make the constructor public

If we're okay we that then I'll do it.

Reply:
Yes, it's what we always do to be able to use makeUnique

Reply:
There's an idiom that addresses this problem: https://playfulprogramming.com/posts/a-forgotten-idiom-revisited-pass-key

It requires a `tag` (what the blog post calls the _pass-key_. The tag is kept private / friend-only, so even if the constructor remains public, clients can't use it and only the `create` method can